appium/webdriver/appium_service.py

```diff
@@ -187,15 +190,15 @@ def stop(self):
             bool: `True` if the service was running before being stopped
         """
         is_terminated = False
-        if self.is_running:
+        if self.is_running and self._process is not None:
```

Comment:
`self._process ` is already checked inside `is_running`

Reply:
Thanks, it was due to below error.

```
appium/webdriver/appium_service.py:194: error: Item "None" of "Optional[Popen[Any]]" has no attribute "terminate"
```

Will add `type: ignore`.
(I guess it's mypy error.)